account-integrations/zkp/relayer/src/services/emailService.ts

```diff
@@ -0,0 +1,127 @@
+import { EventEmitter } from "stream";
+import { Address } from "viem";
+import ImapClient from "../lib/imapClient";
+import EmailTable, { Email, EmailStatus } from "../tables/emailTable";
+import EthereumService from "./ethereumService";
+
+export default class EmailService {
+    constructor(
+        public imapClient: ImapClient,
+        public ethereumService: EthereumService,
+        public emailTable: EmailTable,
+        public pollingInterval: number,
+        public eventEmitter: EventEmitter
+    ) {
+        this.eventEmitter.on("email(s) saved", () => this.processEmails());
+    }
+
+    async start() {
+        await this.pollEmails();
+    }
+
+    async stop() {
+        await this.imapClient.stop();
+    }
+
+    public async pollEmails(): Promise<void> {
+        await this.imapClient.start();
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
```

Comment:
I would instead include some termination flag for this polling loop, such as:
```suggestion
        while (running) {
```
Can set to `true` when `start` is called, and set to `false` when `stop` is called.

Reply:
Follow up https://github.com/getwax/wax/pull/206